great_expectations/datasource/fluent/sql_datasource.py

```diff
@@ -456,7 +456,7 @@ def _add_splitter(self: Self, splitter: Splitter) -> Self:
         # persist the config changes
         context: AbstractDataContext | None
         if context := self._datasource._data_context:
-            context._save_project_config(self._datasource)
+            context.datasources[self._datasource.id] = self._datasource
```

Comment:
Why are we storing this by the id and not the name?

Reply:
Good catch. Addressed in https://github.com/great-expectations/great_expectations/pull/8773/commits/3f228fa3ed30609530ae4d3162336f05d4ad829a